app/js/lib/foxbox/foxbox.js

```diff
@@ -53,15 +53,15 @@ const isSimilar = (objectA, objectB) => {
 };
 
 export default class Foxbox extends Service {
-  constructor() {
+  constructor({ settings, db, net } = { settings: null, db: null, net: null }) {
     super();
 
     // Private properties.
-    this[p.settings] = new Settings();
-    this[p.db] = new Db();
-    this[p.net] = new Network(this[p.settings], (foxboxOnline) => {
-      this._dispatchEvent('box-online', foxboxOnline);
-    });
+    this[p.settings] = settings || new Settings();
+    this[p.db] = db || new Db();
+    this[p.net] = net || new Network(this[p.settings], (foxboxOnline) => {
```

Comment:
__nit:__ weird indent I'd rather do either
```
this[p.net] = net || new Network(this[p.settings], (foxboxOnline) => {
  this._dispatchEvent('box-online', foxboxOnline);
});
```
or
```
this[p.net] = net || 
  new Network(this[p.settings], (foxboxOnline) => {
    this._dispatchEvent('box-online', foxboxOnline);
  });
```

And yeah we should make our IDE's to format in a similar way :)